docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -270,6 +278,10 @@ type MachineDeploymentClass struct {
   // Template is a local struct containing a collection of templates for creation of
   // MachineDeployment objects representing a set of worker nodes.
   Template MachineDeploymentClassTemplate `json:"template"`
+  
+  // MachineHealthCheckClass defines a MachineHealthCheck for this MachineDeploymentClass.
+  // +optional
```

Comment:
```suggestion
  // This field is supported if and only if the control plane provider template
  // referenced above is Machine based and supports setting replicas.
  // +optional
```

Q: Does this restriction of the `machineInfrastructure` also apply to `machineHealthCheck`? 
Q2: If yes, what do we need Machine based **and** replicas, or only Machine based?

Reply:
Based on the implementation PR I would assume we have the same requirements (Machine based and replicas)

Reply:
Is the comment in the wrong spot? Isn't the field above the MachineDeployment one?

Reply:
Absolutely, probably mis-mapped it when reviewing via Intellij

Reply:
Discussed this with @sbueringer and I've added the above comment into L258-L260 i.e. in the MachineHealthCheckClass for the ControlPlane